pipeline/metadata/flatten_hyperquack.py

```diff
@@ -113,35 +114,35 @@ def _process_hyperquack_v1(self, filename: str, scan: Any,
       else:
         domain = sent_domain
 
-      row = {
-          'domain': domain,
-          'category': self.category_matcher.get_category(domain, is_control),
-          'ip': scan['Server'],
-          'date': date,
-          'start_time': result['StartTime'],
-          'end_time': result['EndTime'],
-          'anomaly': scan['Blocked'],
-          'success': result['Success'],
-          'stateful_block': scan['StatefulBlock'],
-          'is_control': is_control,
-          'controls_failed': scan['FailSanity'],
-          'measurement_id': random_measurement_id,
-          'source': flatten_base.source_from_filename(filename),
-      }
+      row = HyperquackRow(
+          domain=domain,
+          category=self.category_matcher.get_category(domain, is_control),
+          ip=scan['Server'],
+          date=date,
+          start_time=result['StartTime'],
+          end_time=result['EndTime'],
+          anomaly=scan['Blocked'],
+          success=result['Success'],
+          stateful_block=scan['StatefulBlock'],
+          is_control=is_control,
+          controls_failed=scan['FailSanity'],
+          measurement_id=random_measurement_id,
+          source=flatten_base.source_from_filename(filename))
 
       if 'Received' in result:
         received = result.get('Received', '')
         received_fields = flatten_base.parse_received_data(
             self.blockpage_matcher, received, scan['Blocked'])
-        row.update(received_fields)
+        row.update_received(received_fields)
```

Comment:
You only need this update because you are using inheritance instead of composition.
Why don't you have a `received` field that contains the entire object?
```suggestion
        row.received = received_fields
```